model/remote_cluster_test.go

```diff
@@ -50,9 +56,90 @@ func TestRemoteClusterIsValid(t *testing.T) {
 func TestRemoteClusterPreSave(t *testing.T) {
 	now := GetMillis()
 
-	o := RemoteCluster{Id: NewId(), ClusterName: "test"}
+	o := RemoteCluster{RemoteId: NewId(), DisplayName: "test"}
 	o.PreSave()
 
 	require.GreaterOrEqual(t, o.CreateAt, now)
-	require.GreaterOrEqual(t, o.LastPingAt, now)
+}
+
+func TestRemoteClusterMsgJson(t *testing.T) {
+	o := RemoteClusterMsg{Id: NewId(), CreateAt: GetMillis(), Token: NewId(), Topic: "shared_channel"}
+
+	json, err := json.Marshal(o)
+	require.NoError(t, err)
+
+	ro, err := RemoteClusterMsgFromJSON(strings.NewReader(string(json)))
+	require.Nil(t, err)
+
+	require.Equal(t, o.Id, ro.Id)
+	require.Equal(t, o.CreateAt, ro.CreateAt)
+	require.Equal(t, o.Token, ro.Token)
+	require.Equal(t, o.Topic, ro.Topic)
+}
+
+func TestRemoteClusterMsgIsValid(t *testing.T) {
+	id := NewId()
+	now := GetMillis()
+	data := []struct {
+		name  string
+		msg   *RemoteClusterMsg
+		valid bool
+	}{
+		{name: "Zero value", msg: &RemoteClusterMsg{}, valid: false},
+		{name: "Missing remote id", msg: &RemoteClusterMsg{Id: id}, valid: false},
+		{name: "Missing Token", msg: &RemoteClusterMsg{Id: id}, valid: false},
+		{name: "Missing Topic", msg: &RemoteClusterMsg{Id: id, Token: NewId()}, valid: false},
+		{name: "RemoteClusterMsg valid", msg: &RemoteClusterMsg{Id: id, Token: NewId(), CreateAt: now, Topic: "shared_channel"}, valid: true},
+	}
+
+	for _, item := range data {
+		err := item.msg.IsValid()
+		if item.valid {
+			assert.Nil(t, err, item.name)
+		} else {
+			assert.NotNil(t, err, item.name)
+		}
+	}
+}
+
+func TestFixTopics(t *testing.T) {
+	testData := []struct {
+		topics   string
+		expected string
+	}{
+		{topics: "", expected: ""},
+		{topics: "   ", expected: ""},
+		{topics: "share", expected: " share "},
+		{topics: "share incident", expected: " share incident "},
+		{topics: " share incident ", expected: " share incident "},
+		{topics: "    share     incident    ", expected: " share incident "},
+	}
+
+	for _, tt := range testData {
+		rc := &RemoteCluster{Topics: tt.topics}
+		rc.fixTopics()
+		assert.Equal(t, tt.expected, rc.Topics)
+	}
+}
+
+func TestRemoteClusterInviteEncryption(t *testing.T) {
+	testData := []struct {
+		name     string
+		password string
+		invite   RemoteClusterInvite
+	}{
+		{name: "empty password", password: "", invite: RemoteClusterInvite{RemoteId: NewId(), SiteURL: "https://example.com:8065", Token: NewId()}},
+		{name: "good password", password: "Ultra secret password!", invite: RemoteClusterInvite{RemoteId: NewId(), SiteURL: "https://example.com:8065", Token: NewId()}},
+	}
```

Comment:
Let's also add a failing case.

Reply:
The only thing that can fail with `Encrypt` is if Linux kernel does not have enough entropy yet to fulfill the random number generator, or the run-time doesn't support AES256.  Neither case will every happen.

I did add a fail test for `Decrypt`.